apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -267,198 +260,76 @@ func (m *ModelManagerAPI) createModelNew(
-		}
-		if !canAddModel {
-			return result, apiservererrors.ErrPerm
-		}
-	}
-
-	ownerTag, err := names.ParseUserTag(args.OwnerTag)
+	svc, err := m.domainServicesGetter.DomainServicesForModel(ctx, modelUUID)
 	if err != nil {
 		return result, errors.Trace(err)
 	}
-
-	// a special case of ErrPerm will happen if the user has add-model permission but is trying to
-	// create a model for another person, which is not yet supported.
-	if !m.isAdmin && ownerTag != m.apiUser {
-		return result, errors.Annotatef(apiservererrors.ErrPerm, "%q permission does not permit creation of models for different owners", permission.AddModelAccess)
-	}
-
-	cloud, err := m.cloudService.Cloud(ctx, cloudTag.Id())
+	newConfig, err := svc.Config().ModelConfig(ctx)
 	if err != nil {
-		if errors.Is(err, errors.NotFound) && args.CloudTag != "" {
-			// A cloud was specified, and it was not found.
-			// Annotate the error with the supported clouds.
-			clouds, err := m.cloudService.ListAll(ctx)
-			if err != nil {
-				return result, errors.Trace(err)
-			}
-			cloudNames := make([]string, 0, len(clouds))
-			for _, cld := range clouds {
-				cloudNames = append(cloudNames, cld.Name)
-			}
-			sort.Strings(cloudNames)
-			return result, errors.NewNotFound(err, fmt.Sprintf(
-				"cloud %q not found, expected one of %q",
-				cloudTag.Id(), cloudNames,
-			))
-		}
-		return result, errors.Annotate(err, "getting cloud definition")
-	}
-
-	var cloudCredentialTag names.CloudCredentialTag
-	if args.CloudCredentialTag != "" {
-		var err error
-		cloudCredentialTag, err = names.ParseCloudCredentialTag(args.CloudCredentialTag)
-		if err != nil {
-			return result, errors.Trace(err)
-		}
-	} else {
-		if ownerTag == controllerModel.Owner() {
-			cloudCredentialTag, _ = controllerModel.CloudCredentialTag()
-		} else {
-			// TODO(axw) check if the user has one and only one
-			// cloud credential, and if so, use it? For now, we
-			// require the user to specify a credential unless
-			// the cloud does not require one.
-			var hasEmpty bool
-			for _, authType := range cloud.AuthTypes {
-				if authType != jujucloud.EmptyAuthType {
-					continue
-				}
-				hasEmpty = true
-				break
-			}
-			if !hasEmpty {
-				return result, errors.NewNotValid(nil, "no credential specified")
-			}
-		}
+		return result, errors.Annotate(err, "getting config")
 	}
 
-	// createModelNew represents the logic needed for moving to DQlite. It is in
-	// a half finished state at the moment for the purpose of removing the model
-	// manager service. This check will go in the very near future.
-	// We check here if the modelService is nil. If it is then we are in testing
-	// mode and don't make the calls so test can keep passing.
-	// THIS IS VERY TEMPORARY.
-	var modelUUID coremodel.UUID
-	if m.modelService != nil {
-		args.CloudRegion = cloudRegionName
-		modelUUID, err = m.createModelNew(ctx, args)
-		if err != nil {
-			return result, err
-		}
-
+	modelInfo, err := m.getModelInfo(ctx, modelTag, true)
```

Comment:
This call is complete over kill now. It gathers up most of the world and we don't use it. A simple call to the model info service above will get the information that we need and no more.

Reply:
No, I changed the getModelInfo only return the information we need.
And we should share getModelInfo with ModelInfo() because they are returning the similar format of data.

Reply:
This method doesn't return apps, machines, but only returns key model info.
We should not change the facade output because this is a client facade.